frontend/src/pages/lesson.js

```diff
@@ -1,22 +1,26 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
 import Container from "react-bootstrap/Container";
+import { useParams } from "react-router-dom";
 import QuizPanel from "../components/quizPanel";
+import { LessonContext } from "../utils/lessonContext";
 
 const LessonQuiz = () => {
-  const filler = {
-    word: "じゃね",
-    choice_0: "goodbye",
-    choice_1: "thank you",
-    choice_2: "maybe",
-    choice_3: "you",
-  };
-  return (
+  const { id } = useParams();
+  const { lesson, setLessonId } = useContext(LessonContext);
+
+  useEffect(() => {
+    setLessonId(id);
+  });
```

Comment:
```suggestion
  const { lesson } = useContext(LessonContext);
```
the removed codes can be implemented in the context. so you dont need some logic here, process it all in the context. 
What you need to do in this page is just get data from context